### PR TITLE
fix scm history going out of sync after commit

### LINKED ISSUE #2922: Git extension widgets "Git" and "Git History" not synchronized
When committing code using the Git extension on Theia, the _Git_ widget is used. After successfully committing, the second widget _Git History_ is not updated. I believe this is not intended  behaviour and thought I therefore mention it here. The current solution for that is to refresh the browser.

If it matters:

- operating system: Ubuntu 18.04
- node: v8.12.0
- npm: 6.4.1
- yarn: 1.9.4
- Google Chrome: Version 69.0.3497.92

### PR DESCRIPTION
**Pull request by erezmus**
#### What it does
- Fixes #2922 issue where new git commits don't appear in the history view
- Also fixes duplicate commits appearing in the history view

#### How to test
- open a git repository
- change a file and commit it
- switch to the history view
- the commit should appear at the top

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by vince-fugnitto — COMMENTED** [review]
@erezmus thank you for the contribution, can you confirm if you are also seeing this bug which I experienced with the pull-request:

- start the application with a workspace under git version control (I was on master)
- make some changes
- commit the changes
- open the history-view and the new commit is present
- in the terminal perform a hard reset (ex: `git reset --hard origin/master`)
- the history-view will fail to show any history

Screen Shot 2023-06-16 at 8 37 18 AM

**Comment by erezmus** [comment]
@vince-fugnitto correct, i can reproduce it. will fix

**Comment by erezmus** [comment]
Pushed a new fix and also added caching of avatars.

**Review by vince-fugnitto — APPROVED** [review]
The updated changes look good to me :+1: 
Do you mind force pushing to restart the `eca check` (it seems to be hanging and is required).

File: packages/scm-extra/src/browser/history/scm-history-widget.tsx [hunk 1]
```diff
@@ -189,19 +188,17 @@ export class ScmHistoryWidget extends ScmNavigableListWidget<ScmHistoryListNode>
...
 
-                    if (options && ((options.maxCount && history.length < options.maxCount) || (!options.maxCount && currentCommits))) {
+                    if (options && (options.maxCount && history.length < options.maxCount)) {
                         this.hasMoreCommits = false;
                     }
-                    if (currentCommits.length > 0) {
-                        history = history.slice(1);
-                    }
+
+                    const avatarCache = new Map<string, string>();
```
Comment (msujew) [inline packages/scm-extra/src/browser/history/scm-history-widget.tsx:200]: Your code never calls `set` on this cache. Is it an oversight, or am I missing something?

**Pull request closed without merging**